### ISSUE TITLE
caching flags?

### ISSUE THREAD

**Issue by mwotton** [commercialhaskell/stack#3156]
this is more a general question - is there a strong reason not to include a hash of the flags (and other relevant configuration) in the cache directory? it's kind of annoying that running a profiling build means rebuilding everything, or adopting a filthy hack like http://www.shimweasel.com/2017/04/27/haskell-profiling-without-pain

even nix doesn't solve this problem for the actual build as far as I'm aware, only for dependencies. Is this something the stack devs think is worth solving? should it be done by another tool?

**Comment by mwotton** [comment]
alternatively, if there's a way to dump all the things stack uses to decide whether something is dirty or not, i'm happy to roll a wrapper script.

**Comment by bitemyapp** [comment]
https://github.com/commercialhaskell/stack/issues/2957 possibly relevant?

**Comment by mgsloan** [comment]
Yup, I've certainly been frustrated with that too at times @mwotton .  I've added a comment to https://github.com/commercialhaskell/stack/issues/2684 , closing this ticket as a duplicate of that
